scripts/import_from_ena_legacy.py

```diff
@@ -0,0 +1,849 @@
+        CACHED_MATERIAL[biosample_id]['source'] = 'biosample server material key'
+
+    parent_animal_list = list()
+    if 'relationships' in data:
+        relationships = data['relationships']
+        for relationship in relationships:
+            # child of relationship only exists in animals
+            if relationship['type'] == 'child of':
+                material_value = {
+                    'text': 'organism',
+                    'ontologyTerms': 'http://purl.obolibrary.org/obo/OBI_0100026'
+                }
+                tmp = {
+                    'accession': biosample_id,
+                    'confirmed': True,
+                    'material': material_value,
+                    'source': 'child of relationship'
+                }
+                CACHED_MATERIAL[biosample_id] = tmp
+                if relationship['target'] != biosample_id:
+                    parent_animal_list.append(relationship['target'])
+            elif relationship['type'] == 'derived from':
+                if relationship['target'] != biosample_id:
+                    # try to get all records from the relationships
+                    if relationship['target'] not in CACHED_MATERIAL:
+                        retrieve_biosamples_record(es, es_index_prefix, relationship['target'])
+                    related_material = CACHED_MATERIAL[relationship['target']]
+                    if 'material' in related_material and related_material['confirmed'] \
+                            and related_material['material']['text'] == 'organism':
+                        if biosample_id in CACHED_MATERIAL and not CACHED_MATERIAL[biosample_id]['confirmed']:
+                            material_value = {
+                                'text': 'specimen from organism',
+                                'ontologyTerms': 'http://purl.obolibrary.org/obo/OBI_0001479'
+                            }
+                            tmp = {
+                                'accession': biosample_id,
+                                'confirmed': True,
+                                'material': material_value,
+                                'source': 'derived from organism relationship'
+                            }
+                            CACHED_MATERIAL[biosample_id] = tmp
+                        animal = relationship['target']
+
+    if biosample_id not in CACHED_MATERIAL:
+        CACHED_MATERIAL.setdefault(biosample_id, {})
+        CACHED_MATERIAL[biosample_id]['accession'] = biosample_id
+        CACHED_MATERIAL[biosample_id]['confirmed'] = False
+        CACHED_MATERIAL[biosample_id]['material'] = default_material_value
+
+    es_doc = dict()
+    # record labelled as FAANG does not guarantee to be in ES as it may fail the validation,
+    # but should not be dealt with here
+
+    if 'project' in data['characteristics'] and data['characteristics']['project'][0]['text'] == 'FAANG':
+        pass
```

Comment:
Should add log info or remove this statement at all, as currently it doesn't have any effect.

Reply:
done in 27a573a